core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFileSupplier.java

```diff
@@ -0,0 +1,159 @@
+     * disk file is unchanged will return the same object.
+     *
+     * @param path Path to the file to read from.
+     * @return Implementation of {@link ProfileFileSupplier} that is capable of supplying a new profile when the file
+     *         has been modified.
+     */
+    static ProfileFileSupplier reloadWhenModified(Path path) {
+        return new ProfileFileSupplier() {
+
+            final ProfileFile.Builder builder = ProfileFile.builder()
+                                                           .content(path)
+                                                           .type(ProfileFile.Type.CREDENTIALS);
+
+            final ProfileFileRefresher refresher = ProfileFileRefresher.builder()
+                                                                       .profileFile(builder::build)
+                                                                       .profileFilePath(path)
+                                                                       .build();
+
+            @Override
+            public ProfileFile get() {
+                return refresher.refreshIfStale();
+            }
+
+            @Override
+            public void close() {
+                refresher.close();
+            }
+        };
+    }
+
+    /**
+     * Creates a {@link ProfileFileSupplier} capable of producing a single profile object from a file.
+     *
+     * @param path Path to the file to read from.
+     * @return Implementation of {@link ProfileFileSupplier} that is capable of supplying a single profile.
+     */
+    static ProfileFileSupplier fixedProfileFile(Path path) {
+        ProfileFile profileFile = ProfileFile.builder()
+                                             .content(path)
+                                             .type(ProfileFile.Type.CREDENTIALS)
+                                             .build();
+
+        return () -> profileFile;
+    }
+
+    /**
+     * Creates a {@link ProfileFileSupplier} that produces an existing profile.
+     *
+     * @param profileFile Profile object to supply.
+     * @return Implementation of {@link ProfileFileSupplier} that is capable of supplying a single profile.
+     */
+    static ProfileFileSupplier fixedProfileFile(ProfileFile profileFile) {
+        return () -> profileFile;
+    }
+
+    /**
+     * creates a {@link ProfileFileSupplier} that produces an existing non-null profile. If the given profile
+     * is null, then the created supplier will also be null.
+     *
+     * @param profileFile Profile object to supply.
+     * @return Implementation of {@link ProfileFileSupplier} that is capable of supplying a single profile.
+     */
+    static ProfileFileSupplier wrapIntoNullableSupplier(ProfileFile profileFile) {
+        return Objects.nonNull(profileFile) ? () -> profileFile : null;
+    }
+
+    /**
+     * A builder for {@link ProfileFileSupplier}.
+     */
+    interface Builder {
```

Comment:
Can we move this interface out?

Reply:
Since this is an internal API, removed interface completely. Only left builder implementation.